mk/config.mk

```diff
@@ -599,11 +599,11 @@ CFG_SCMI_MSG_RESET_DOMAIN ?= n
 CFG_SCMI_MSG_SMT ?= n
 
 ifneq ($(CFG_STMM_PATH),)
-$(call force,CFG_WITH_SECURE_PARTITION,y)
+$(call force,CFG_WITH_STMM_SP,y)
```

Comment:
Changing the configuration will have an effect on all users trying to build the StMM partition, and will break their builds.
Please sync-up with @apalos  and the OP-TEE maintainers to see who we shall notify and align-up with.

Reply:
I think that's fine here as long as the CFG_WITH_SECURE_PARTITION is changed accordingly in optee core. 
The user still has to define CFG_STMM_PATH=/path/to/BL32_AP_MM.fd so nothing changes compilation wise. Right?

Reply:
Correct, if the user uses CFG_STMM_PATH then CFG_WITH_STMM_SP is set instead of CFG_WITH_SECURE_PARTITION.  I also changed the ifdef and IS_ENABLED from  CFG_WITH_SECURE_PARTITION to  CFG_WITH_STMM_SP

Reply:
Please add this clarification to the commit message.

Reply:
Done